File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPOffHand.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPOffHand implements PlayerStateProperty<ItemStack> {
    @Override
    public String getConfigName() {
        return "off_hand";
    }

    @Override
    public ItemStack getDefaultSerializedProperty() {
        return new ItemStack(Material.AIR);
    }

    @Override
    public ItemStack getSerializedPropertyFromPlayer(Player player) {
        return player.getInventory().getItemInOffHand();
    }

    @Override
    public void restoreProperty(Player player, ItemStack offHand) {
        player.getInventory().setItemInOffHand(offHand);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPOffHand.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPOffHand extends PlayerStateProperty[ItemStack]:
  override def getConfigName: String = "off_hand"

  override def getDefaultSerializedProperty: ItemStack =
    ItemStack(Material.AIR)

  override def getSerializedPropertyFromPlayer(player: Player): ItemStack =
    player.getInventory.getItemInOffHand

  override def restoreProperty(player: Player, offHand: ItemStack): Unit =
    player.getInventory.setItemInOffHand(offHand)